stdlib/io/src/main/ballerina/src/io/readable_character_channel.bal

```diff
@@ -26,34 +27,68 @@ public type ReadableCharacterChannel object {
     # + channel - ReadableByteChannel which would be used to read characters
     # + charset - Character-Set which would be used to encode/decode given bytes to characters
     public function __init(ReadableByteChannel byteChannel, string charset) {
-        self.init(byteChannel, charset);
+        initReadableCharacterChannel(self, byteChannel, java:fromString(charset));
     }
 
-    # Initializes a character ReadableCharacterChannel.
-    #
-    # + bChannel - ReadableByteChannel which should be used to initialize the ReadableCharacterChannel
-    # + cs - Character-set (i.e UTF-8) which should be used to encode/decode
-    function init(ReadableByteChannel bChannel, string cs) = external;
-
     # Reads a given number of characters. This will attempt read up to `numberOfChars` characters from the channel.
     # `io:EofError` will return once channel reach to it end.
     #
     # + numberOfChars - Number of characters which should be read
     # + return - Content which is read or `EofError` once channel reach to it end. `Error` if any error occurred.
-    public function read(@untainted int numberOfChars) returns @tainted string|Error = external;
+    public function read(@untainted int numberOfChars) returns @tainted string|Error {
+        handle|Error result = readExtern(self, numberOfChars);
+        if (result is handle) {
+            return <string>java:toString(result);
```

Comment:
Shall we create an `Error` and return it if the `toString()` function returns `nil`?

Otherwise, if we get a `nil` value, it panics.

Reply:
Internally this will never produce a Java null string. So this toString never produces an nill value.